dbctx/soup.go

```diff
@@ -0,0 +1,83 @@
+package dbctx
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+type (
+	soupRepository struct {
+		connFactory ConnectionFactory
+	}
+	// SoupRepository to provide access to methods to handle soup data
+	SoupRepository interface {
+		// SetSoup sets the soup of the day
+		SetSoup(name string) error
+		GetSoupOfTheDay() (*SoupOfTheDay, error)
+	}
+)
+
+// NewSoupRepository creates a new repository
+func NewSoupRepository(connFactory ConnectionFactory) SoupRepository {
+	return &soupRepository{
+		connFactory,
+	}
+}
+
+func (m *soupRepository) SetSoup(name string) error {
+	db, err := m.connFactory.Open()
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	tx := db.Begin()
+
+	soupOfTheDay := getCreateSoupOfTheDay(tx)
+	soupOfTheDay.Soup = *getSoup(name, tx)
+
+	tx.Save(soupOfTheDay)
+
+	tx.Commit()
+	return tx.Error
+}
+
+func (m *soupRepository) GetSoupOfTheDay() (*SoupOfTheDay, error) {
+	db, err := m.connFactory.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	tx := db.Begin()
+	soupOfTheDay := getCreateSoupOfTheDay(tx)
+	tx.Commit()
+	return soupOfTheDay, tx.Error
+}
+
+// getCreateSoupOfTheDay returns the soup of the day, pass DB or transaction to be used to extract the data.
+func getCreateSoupOfTheDay(tx *gorm.DB) *SoupOfTheDay {
```

Comment:
Is it on purpose that you don't have this as a method on the struct? I'm considering whether there's a use-case for not doing so. The only thing I can come up with right now is that other parts of the package could call `getCreateSoupOfTheDay` without having a `soupRepository`